### ISSUE TITLE
Expose Prometheus TSDB compaction metrics

### ISSUE THREAD

**Issue by pracucci** [cortexproject/cortex#2174]
The following TSDB metrics are not exposed by Cortex (blocks storage) but are actually quite important to be able to graph at which point in time TSDB head compactions occur:

```
prometheus_tsdb_compactions_triggered_total
prometheus_tsdb_compactions_failed_total
```

We should add them to the mapping in `pkg/ingester/metrics.go` and update unit tests accordingly.

**Comment by pstibrany** [comment]
I planned to include this in https://github.com/cortexproject/cortex/pull/2172, but forgot. As I'm going to do some changes in that PR, I will include it.

**Comment by pstibrany** [comment]
Do you think that it would be interesting to include `prometheus_tsdb_compactions_skipped_total` ("Total number of skipped compactions due to disabled auto compaction."), since we are disabling auto compaction now? I'm thinking that even when we run compaction manually, this value shouldn't get very high, otherwise settings are not correct.

**Comment by pracucci** [comment]
Why the value of the skipped will not go high? If we try the compact every
minute but the block range period is 2h, most of them will be skipped. Am I
missing something?

**Comment by pstibrany** [comment]
> Why the value of the skipped will not go high? If we try the compact every minute but the block range period is 2h, most of them will be skipped. Am I missing something?

OK, looking at the code, I think misunderstood head compaction – TSDB code indeed looks like it tries to do it every minute (!?), or on appender's commit under certain conditions – so it would get high pretty quickly.

**Comment by pstibrany** [comment]
> ```
> prometheus_tsdb_compactions_triggered_total
> prometheus_tsdb_compactions_failed_total
> ```

Turns out these are actually not very interesting. TSDB tries to "trigger" compaction all the time, but since autocompactions are disabled, nothing happens.

I think we should count these internally in Cortex, and increase them when we actually run compaction, and on failures.

**Comment by pstibrany** [comment]
> I think we should count these internally in Cortex, and increase them when we actually run compaction, and on failures.

This is what I did in #2172.

**Comment by pstibrany** [comment]
This was fixed by #2172

**Comment by pstibrany** [comment]
```
cortex_ingester_tsdb_compactions_triggered_total
cortex_ingester_tsdb_compactions_failed_total
```
